### DISCUSSION TITLE
Show extension addon

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#1715] [Discussion]
I think it would be more coherent if the selectors of the extensions, when automatically added to the category list, would be _above_ the My Blocks selector instead of being below.

**Comment by retronbv** [comment]
Can you explain more? If i understand correctly, you want the my blocks category to be under the automatically added extensions?

**Reply by sfederici** [reply]
Yes, exactly. The My Blocks category (when the extensions are added automatically, so to go back to the way they were organized in Scratch 2) should be the last category. If this organization is not appreciated by the Addon developer, this could be chosen by adding a checkbox.

**Reply by retronbv** [reply]
cc @TheColaber (Addon Developer)
I'm sure TheColaber could add a checkbox to change the order?

**Reply by WorldLanguages** [reply]
I don't think this has anything to do with the "auto-show block category" addon? @sfederici Isn't the order you mention the same if you add the extension manually? I think you're requesting an addon to change the order instead?

**Reply by sfederici** [reply]
Right now, if we add extensions manually or automatically, they are added under the My Blocks category. I'd like having a switch in the "auto-show block category" addon to allow users to choose if automatically added category are considered on a par with standard categories, so that they are shown above the My Blocks category. Having a separate addon that allow you to choose in what order to show extensions in the category list would have a slightly different behavior (all extensions would be above or under the My Blocks category, with no difference between "standard" and "extended" categories). Anyway, if think that the addon must be kept separate from the "auto-show block category" addon, I can live with it :)

**Reply by WorldLanguages** [reply]
Yeah, as you said, manually or automatically adding extensions both produce the same order so I'm not sure why this should be part of the "auto-show block category" addon.
Also tbh I'd prefer not to let users reorder categories because many addons (data category tweaks, category icons) expect the order to be the same as vanilla Scratch. But a "custom order block category" addon could be a thing.

**Reply by sfederici** [reply]
Fine for me
